packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -1332,12 +1332,14 @@ describe('Collection aggregations tab', function () {
       });
       await browser.waitForAriaDisabled(previousButton, true);
 
-      // previousButton has a tooltip, to close it we press Escape
-      // and wait a bit (for the debounced close to kick in)
-      await browser.keys([Key.Escape]);
-      await sleep(50);
+      // previousButton has a tooltip and as it becomes disabled,
+      // tooltip disappers. wait for it to disappear
+      await browser.waitForAnimations(Selectors.FocusModeModal);
 
-      // the next Escape is for the modal to close
+      // As the tooltip disappeared at this point, the previousButton still
+      // has the focus. Pressing escape twice, should close the modal.
+      await browser.keys([Key.Escape]);
+      await sleep(100);
       await browser.keys([Key.Escape]);
```

Comment:
Maybe there is a way for us to do it without an arbitrary timeout, like removing the focus from the button? Also just generally wondering why button is swallowing the Esc keypress in this case, is this a bug?

Reply:
i think without complicating it even more, i'd do as we do at other places, close the modal using its close button.